airbyte-server/src/main/java/io/airbyte/server/ServerApp.java

```diff
@@ -214,19 +214,22 @@ public static void main(String[] args) throws Exception {
 
     Optional<String> airbyteDatabaseVersion = jobPersistence.getVersion();
     boolean isKubernetes = configs.getWorkerEnvironment() == WorkerEnvironment.KUBERNETES;
-    if (!isKubernetes && airbyteDatabaseVersion.isPresent() && !AirbyteVersion.isCompatible(airbyteVersion, airbyteDatabaseVersion.get())
+    if (airbyteDatabaseVersion.isPresent() && !AirbyteVersion.isCompatible(airbyteVersion, airbyteDatabaseVersion.get())
```

Comment:
What's the difference between incompatible Airbyte Version and the Database Version being ahead of the App version?

Reply:
There are two scenarios :
1. Database version is before the App version, in this case people are trying to upgrade and automatic migration can run
2. Database version is ahead of App version, in this case people are trying to downgrade and automatic migration can not run

Reply:
can't we combine `AirbyteVersion.isCompatible` and ` isDatabaseVersionAheadOfAppVersion`? 

`isDatabaseVersionAheadOfAppVersion` seems to be more strict check of `AirbyteVersion.isCompatible`. why have both?

Reply:
Am not sure how to combine these two. Both tell us different things, `AirbyteVersion.isCompatible` is not just used for automatic migration but other places while `isDatabaseVersionAheadOfAppVersion` tells us whether we can run auto migration or not.

Reply:
Hmm I should be clearer, I meant to get rid of the `isCompatible` check, since the condition it's checking for must be true to fulfill `sDatabaseVersionAheadOfAppVersion`'s intent.

what about something like this?

```
     if (airbyteDatabaseVersion.isPresent() && isDatabaseVersionBehindAppVersion(airbyteVersion, 
     airbyteDatabaseVersion.get())) {
     ....

  private static boolean isDatabaseVersionBehindAppVersion(String airbyteVersion, String airbyteDatabaseVersion) {
    AirbyteVersion serverVersion = new AirbyteVersion(airbyteVersion);
    AirbyteVersion databaseVersion = new AirbyteVersion(airbyteDatabaseVersion);

    return databaseVersion.getMinorVersion().compareTo(serverVersion.getMinorVersion()) < 0;
  }
```

we can simplify by getting rid of the not conditional, and also simplify the logic of `isDatabaseVersionBehindAppVerion`. wdyt?